components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java

```diff
@@ -228,9 +237,17 @@ public boolean enrollDevice(Device device) throws DeviceManagementException {
                         }
                     } catch (DeviceManagementDAOException e) {
                         DeviceManagementDAOFactory.rollbackTransaction();
-                        throw new DeviceManagementException("Error occurred while adding enrolment related metadata", e);
+                        String msg = "Error occurred while adding enrolment related metadata";
+                        log.error(msg, e);
+                        throw new DeviceManagementException(msg, e);
                     } catch (TransactionManagementException e) {
-                        throw new DeviceManagementException("Error occurred while initiating transaction", e);
+                        String msg = "Error occurred while initiating transaction";
+                        log.error(msg);
+                        throw new DeviceManagementException(msg, e);
+                    } catch (Exception e) {
```

Comment:
Please remove this catch block as there won't be any cases where we hit this in normal flow. Do the same for all other places.

Reply:
Please ignore this comment.